pkg/providers/launchtemplate/testdata/al2_userdata_merged.golden

```diff
@@ -12,8 +12,7 @@ Content-Type: text/x-shellscript; charset="us-ascii"
 
 #!/bin/bash -xe
```

Comment:
I could go either way but is it worth just creating an entirely separate file for the new version of the generated userData so that a reader doesn't have to reason about more template params

Reply:
I was on the fence about this as well.  I liked this approach since it could be expanded to deal with other version dependent / optional arguments without needing a file for each combination. Since that's not the case now though I'd agree that multiple files may be the better way to go from a readability standpoint.